x/inflation/handler.go

```diff
@@ -0,0 +1,24 @@
+package inflation
+
+import (
+	errorsmod "cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/evmos/evmos/v10/x/inflation/types"
+)
+
+// NewHandler returns a handler for Inflation type messages.
+func NewHandler(server types.MsgServer) sdk.Handler {
+	return func(ctx sdk.Context, msg sdk.Msg) (result *sdk.Result, err error) {
+		ctx = ctx.WithEventManager(sdk.NewEventManager())
+
+		switch msg := msg.(type) {
+		case *types.MsgUpdateParams:
+			res, err := server.UpdateParams(ctx, msg)
+			return sdk.WrapServiceResult(ctx, res, err)
+		default:
+			err := errorsmod.Wrapf(errors.ErrUnknownRequest, "unrecognized %s message type: %T", types.ModuleName, msg)
```

Comment:
Nit: refactor if above suggestion is committed
```suggestion
			err := errorsmod.Wrapf(errortypes.ErrUnknownRequest, "unrecognized %s message type: %T", types.ModuleName, msg)
```